mods/nyancat/init.lua

```diff
@@ -0,0 +1,95 @@
+minetest.register_node("nyancat:nyancat", {
+	description = "Nyan Cat",
+	tiles = {"default_nc_side.png", "default_nc_side.png", "default_nc_side.png",
+		"default_nc_side.png", "default_nc_back.png", "default_nc_front.png"},
+	paramtype2 = "facedir",
+	groups = {cracky = 2},
+	is_ground_content = false,
+	legacy_facedir_simple = true,
+	sounds = default.node_sound_defaults(),
+})
+
+minetest.register_node("nyancat:nyancat_rainbow", {
+	description = "Nyan Cat Rainbow",
+	tiles = {
+		"default_nc_rb.png^[transformR90", "default_nc_rb.png^[transformR90",
+		"default_nc_rb.png", "default_nc_rb.png"
+	},
+	paramtype2 = "facedir",
+	groups = {cracky = 2},
+	is_ground_content = false,
+	sounds = default.node_sound_defaults(),
+})
+
+minetest.register_craft({
+	type = "fuel",
+	recipe = "nyancat:nyancat",
+	burntime = 1,
+})
+
+minetest.register_craft({
+	type = "fuel",
+	recipe = "nyancat:nyancat_rainbow",
+	burntime = 1,
+})
+
+nyancat = {}
+
+function nyancat.place(pos, facedir, length)
+	local tailvec = {x = 0, y = 0, z = 0}
+	if facedir == 0 then
+		tailvec.z = 1
+	elseif facedir == 1 then
+		tailvec.x = 1
+	elseif facedir == 2 then
+		tailvec.z = -1
+	elseif facedir == 3 then
+		tailvec.x = -1
+	else
+		facedir = 0
+		tailvec.z = 1
+	end
+	local p = {x = pos.x, y = pos.y, z = pos.z}
+	minetest.set_node(p, {name = "nyancat:nyancat", param2 = facedir})
+	for i = 1, length do
+		p.x = p.x + tailvec.x
+		p.z = p.z + tailvec.z
+		minetest.set_node(p, {name = "nyancat:nyancat_rainbow", param2 = facedir})
+	end
+end
+
+function nyancat.generate(minp, maxp, seed)
+	local height_min = -31000
+	local height_max = -32
+	if maxp.y < height_min or minp.y > height_max then
+		return
+	end
+	local y_min = math.max(minp.y, height_min)
+	local y_max = math.min(maxp.y, height_max)
+	local volume = (maxp.x - minp.x + 1) * (y_max - y_min + 1) * (maxp.z - minp.z + 1)
+	local pr = PseudoRandom(seed + 9324342)
+	local max_num_nyancats = math.floor(volume / (16 * 16 * 16))
+	for i = 1, max_num_nyancats do
+		if pr:next(0, 1000) == 0 then
+			local x0 = pr:next(minp.x, maxp.x)
+			local y0 = pr:next(minp.y, maxp.y)
+			local z0 = pr:next(minp.z, maxp.z)
+			local p0 = {x = x0, y = y0, z = z0}
+			nyancat.make(p0, pr:next(0, 3), pr:next(3, 15))
+		end
+	end
+end
+
+if minetest.get_mapgen_setting("mg_name") ~= "singlenode" then
```

Comment:
As suggested in https://github.com/minetest/minetest_game/pull/1190#issuecomment-231827129 please can this line be removed? As it has no effect anyway and nyancats will often be wanted in singlenode.